charms/jupyter-ui/tests/integration/test_charm.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+
+"""Integration tests for Jupyter UI Operator/Charm."""
+
+import logging
+from pathlib import Path
+
+import aiohttp
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CONFIG = yaml.safe_load(Path("./config.yaml").read_text())
+APP_NAME = "jupyter-ui"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build and deploy the charm.
+
+    Assert on the unit status.
+    """
+    charm_under_test = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+
+    # NOTE: idle_period is used to ensure all resources are deployed
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10, idle_period=30
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+
+
+async def fetch_response(url):
+    """Fetch provided URL and return pair - status and text (int, string)."""
+    result_status = 0
+    result_text = ""
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as response:
+            result_status = response.status
+            result_text = await response.text()
+    return result_status, str(result_text)
+
+
+async def test_ui_is_accessible(ops_test: OpsTest):
+    """Verify that UI is accessible."""
+    # NOTE: This test is re-using deployment created in test_build_and_deploy()
+    # NOTE: This test also tests Pebble checks since it uses the same URL.
```

Comment:
Can you elaborate on the pebble checks part?  I dont see how this is asserting on pebble checks

Reply:
Pebble checks are defined in Pebble layer as:
```
         "checks": {
                "up": {
                    "override": "replace",
                    "period": "30s",
                    "http": {"url": f"http://localhost:{self._http_port}"},
                },
            },
```
The URL in Pebble checks is the same as UI URL. If we fetch that one for UI accessibility test we also test response to Pebble check.
We do not test Pebble fetching that URL, but making sure the URL we give to Pebble is reachable.